opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/RestTemplateTracingAutoConfiguration.java

```diff
@@ -0,0 +1,174 @@
+import io.opentracing.contrib.spring.web.client.TracingAsyncRestTemplateInterceptor;
+import io.opentracing.contrib.spring.web.autoconfig.client.TracingRestTemplateCustomizer;
+import io.opentracing.contrib.spring.web.client.TracingRestTemplateInterceptor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.boot.web.client.RestTemplateCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.AsyncClientHttpRequestInterceptor;
+import org.springframework.http.client.ClientHttpRequestInterceptor;
+import org.springframework.http.client.support.InterceptingAsyncHttpAccessor;
+import org.springframework.http.client.support.InterceptingHttpAccessor;
+import org.springframework.web.client.AsyncRestTemplate;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Instrumentation of {@link RestTemplate} and {@link AsyncRestTemplate}.
+ * <p>
+ * For Spring Boot it also instruments {@link RestTemplateBuilder} and all instances created with it.
+ *
+ * @author Pavol Loffay
+ */
+@Configuration
+@ConditionalOnBean(Tracer.class)
+@ConditionalOnClass(RestTemplate.class)
+@ConditionalOnProperty(prefix = WebClientTracingProperties.CONFIGURATION_PREFIX, name = "enabled", matchIfMissing = true)
+@AutoConfigureAfter(TracerAutoConfiguration.class)
+@EnableConfigurationProperties(WebClientTracingProperties.class)
+public class RestTemplateTracingAutoConfiguration {
+
+    private static final Log log = LogFactory.getLog(RestTemplateTracingAutoConfiguration.class);
+
+    /**
+     * Provides bean {@link RestTemplateSpanDecorator.StandardTags}.
+     */
+    @Configuration
+    @ConditionalOnMissingBean(RestTemplateSpanDecorator.class)
+    public static class StandardTagsConfiguration {
+
+        @Bean
+        public RestTemplateSpanDecorator.StandardTags standardTagsRestTemplateSpanDecorator() {
+            return new RestTemplateSpanDecorator.StandardTags();
+        }
+    }
+
+    /**
+     * Injects {@link TracingRestTemplateInterceptor} into {@link InterceptingHttpAccessor#getInterceptors()}.
+     */
+    @Configuration
+    @ConditionalOnBean(InterceptingHttpAccessor.class)
+    public static class RestTemplatePostProcessingConfiguration {
```

Comment:
Its actual base class that provides interceptor support for RestTemplate. Its questionable, whether limit it to RestTemplate or use this potentionally wider base class.

Conditional on Async... is because its deprecated and will be removed probably in Spring 6.x, therefore ConditionalOnClass makes it forward compatible.